app/services/service_owner_constraint.rb

```diff
@@ -0,0 +1,9 @@
+class ServiceOwnerConstraint
+  def self.matches?(request)
+    id = request.session["warden.user.user.key"].try(:[], 0).try(:[], 0)
```

Comment:
Did you find out if you could do `try(:first).try(:first)` here by the way?

Reply:
Also, do you think being more explicit here and calling this `user_id_from_session` would make this clearer? Just thinking new readers (/mostly me) seeing `"warden.user.user.key"` might be confused about what that is!